plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreConfig.java

```diff
@@ -325,4 +333,31 @@ public ThriftMetastoreConfig setWriteStatisticsThreads(int writeStatisticsThread
         this.writeStatisticsThreads = writeStatisticsThreads;
         return this;
     }
+
+    public String getHttpToken()
+    {
+        return httpToken;
+    }
+
+    @Config("hive.metastore.http.client.bearer-token")
```

Comment:
Some config properties in this class should be disallowed when HTTP interface is used. For instance, this PR doesn't support impersonation if I understand correctly. So, connector initialization should fail when `hive.metastore.thrift.impersonation.enabled` is enabled and `hive.metastore.uri` uses http(s). 

Please fix and add some tests in TestHivePlugin.

Reply:
Thanks for the comment. Can you please provide a pointer on how to do this? The problem as I see it is that `hive.metastore.uri` is defined in `StaticMetastoreConfig` and `hive.metastore.thrift.impersonation.enabled` is defined in `ThriftMetastoreConfig`.

Reply:
See https://github.com/trinodb/trino/blob/325a7e828989542af5d215fb97928123aa422434/lib/trino-hdfs/src/main/java/io/trino/hdfs/gcs/HiveGcsConfig.java#L72-L81 for reference.

Reply:
May be I am missing something. All the fields in the validate method above refer to the fields which in the HiveGcsConfig so that makes sense. However, IIUC this comment wants us to validate that if `hive.metastore.uri` has `http[s]` scheme then `hive.metastore.thrift.impersonation.enabled` should not be set to true. However `hive.metastore.thrift.impersonation.enabled` is defined in `ThriftMetastoreConfig` class and `hive.metastore.uri` is defined in `StaticMetastoreConfig`.  Is there a way to get the config value from one class in another?

Reply:
@vihangk1 good point. Sorry for misleading you.
We have though the option to validate these two configs in `ThriftMetastoreModule`

Use something similar to this code snippet in `ThriftMetastoreModule#setup(Binder)`.

```
        StaticMetastoreConfig staticMetastoreConfig = buildConfigObject(StaticMetastoreConfig.class);
        ThriftMetastoreConfig thriftMetastoreConfig = buildConfigObject(ThriftMetastoreConfig.class);

        if (!staticMetastoreConfig.getMetastoreUris().isEmpty()) {
            boolean isHttpMetastore = staticMetastoreConfig.getMetastoreUris().stream()
                    .anyMatch(uri -> "http".equalsIgnoreCase(uri.getScheme()) || "https".equalsIgnoreCase(uri.getScheme()));

            if (isHttpMetastore && thriftMetastoreConfig.isImpersonationEnabled()){
                throw new IllegalStateException("'hive.metastore.thrift.impersonation.enabled' cannot be set to true while using http/https metastore URIs in 'hive.metastore.uri'");
            }
        }
```